.github/workflows/publish-npm-master.yaml

```diff
@@ -18,25 +18,48 @@ jobs:
     name: "Publish to NPM"
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
         with:
           node-version: "14"
           registry-url: "https://registry.npmjs.org"
-          cache: "npm"
+
       - name: Ensure NPM access
         run: npm whoami
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
-      - run: npm i npm@8 -g # we need a version of npm that supports workspaces
-      - run: npm ci
+
+      - name: Cache PNPM modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.pnpm-store
+          key: ${{ runner.os }}-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-
+
+      - name: Setup PNPM
+        uses: pnpm/action-setup@v2.1.0
+        with:
+          version: 7.0.0
+          run_install: true
+
+      - name: Run ESLint and Build
+        run: |
+          pnpm build
+          pnpm lint
```

Comment:
`lint` should be before `build` right? 

If lint fails it should not build invalid code.